### DISCUSSION TITLE
Random Restarts for Botorch GP models

### DISCUSSION THREAD

**Discussion by huylenguyen** [meta-pytorch/botorch#1717] [Q&A] [answered]
Hello, I am wondering if Botorch has any functionality built in for multiple random restarts when optimising the log-likelihood of the model? I have been told that this optimisation is not necessarily convex, and so it may be useful to perform multiple random runs of the optimizer.

**Comment by saitcakmak** [comment, marked as answer]
Hi @huylenguyen.  We do no have this exposed out of the box.  However, our default model fitting function (`fit_gpytorch_mll`) has a random restart approach that is used to handle any errors that may get raised during model fitting. The logic for this is [here](https://github.com/pytorch/botorch/blob/main/botorch/fit.py#L234). If you wanted to do random restarts and pick the best, you could modify that to record the result of each fitting attempt -- rather than returning the first time model fitting succeeds -- and pick the best model afterwards.

**Reply by Balandat** [reply]
We should probably make it easier for folks to do this at some point. Created #1724 for tracking.
